src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
you can't because xupdate happen on nodes set and `appendChildren` used at different places.

Well, if you like such naming and think that it'll not confuse people ... then I'm 1st confused creature.